samcli/local/docker/utils.py

```diff
@@ -154,3 +155,22 @@ def get_docker_platform(architecture: str) -> str:
     validate_architecture(architecture)
 
     return f"linux/{get_image_arch(architecture)}"
+
+def get_posix_effective_user() -> Dict[str, str]:
+    """
+    Returns the posix effective user and group id
+
+    :returns Dictionary containing user and group id, or None for non posix systems
+    """
+    if os.name.lower() == 'posix':
+        effective_user = {}
```

Comment:
Can we use `@dataclass` instead of dictionary here? It will be easy to use in other places (no need to hardcode its keys for instance).